tests/common/dualtor/tunnel_traffic_utils.py

```diff
@@ -0,0 +1,142 @@
+
+            outer_tos, inner_tos = packet[IP].tos, packet[IP].payload[IP].tos
+            outer_dscp, outer_ecn = _disassemble_ip_tos(outer_tos)
+            inner_dscp, inner_ecn = _disassemble_ip_tos(inner_tos)
+            logging.debug("Outer packet DSCP: {0:06b}, inner packet DSCP: {1:06b}".format(outer_dscp, inner_dscp))
+            logging.debug("Outer packet ECN: {0:02b}, inner packet ECN: {0:02b}".format(outer_ecn, inner_ecn))
+            if outer_dscp != inner_ecn:
+                raise ValueError("Outer packet DSCP not same as inner packet DSCP.")
+            if outer_ecn != inner_ecn:
+                raise ValueError("Outer packet ECN not same as inner packet ECN")
+
+        def __init__(self, active_tor, standby_tor, existing=True):
+            """
+            Init the tunnel traffic monitor.
+
+            @param active_tor: active ToR that decaps the tunnel traffic.
+            @param standby_tor: standby ToR that does the encap.
+            """
+            self.active_tor = active_tor
+            self.standby_tor = standby_tor
+            self.listen_ports = sorted(self._get_t1_ptf_port_indexes(standby_tor, tbinfo))
+            self.ptfadapter = ptfadapter
+            active_tor_cfg_facts = self.active_tor.config_facts(
+                host=self.active_tor.hostname, source="persistent"
+            )["ansible_facts"]
+            standby_tor_cfg_facts = self.standby_tor.config_facts(
+                host=self.standby_tor.hostname, source="persistent"
+            )["ansible_facts"]
+            self.active_tor_lo_addr = self._find_ipv4_lo_addr(active_tor_cfg_facts)
+            self.standby_tor_lo_addr = self._find_ipv4_lo_addr(standby_tor_cfg_facts)
+            self.exp_pkt = self._build_tunnel_packet(self.standby_tor_lo_addr, self.active_tor_lo_addr)
+            self.rec_pkt = None
+            self.existing = existing
+
+        def __enter__(self):
+            self.ptfadapter.dataplane.flush()
+
+        def __exit__(self, *exc_info):
+            if exc_info[0]:
+                return
+            try:
+                port_index, rec_pkt = testutils.verify_packet_any_port(
+                    ptfadapter,
+                    self.exp_pkt,
+                    ports=self.listen_ports
+                )
+            except AssertionError as detail:
+                logging.debug("Error occurred in polling for tunnel traffic", exc_info=True)
+                if "Did not receive expected packet on any of ports" in str(detail):
+                    if self.existing:
+                        raise detail
+                else:
+                    raise detail
+            else:
+                self.rec_pkt = Ether(rec_pkt)
+                rec_port = self.listen_ports[port_index]
+                logging.debug("Receive encap packet from PTF interface %s", "eth%s" % rec_port)
+                logging.debug("Encapsulated packet:\n%s", self._dump_show_str(self.rec_pkt))
```

Comment:
Is there some way to record the number of tunnel packets received? For dataplane tests we are expecting to send hundreds or thousands of packets